### ISSUE TITLE
 Argument 2 passed to Zend\Expressive\Container\ApplicationFactory::injectRoutesAndPipeline() must be of the type array, object given

### ISSUE THREAD

**Issue by Naktibalda** [zendframework/zend-expressive#446]
This error started happening after 1.1.0 release.

https://travis-ci.org/Codeception/Codeception/jobs/201309495#L3141

> PHP Catchable fatal error:  Argument 2 passed to Zend\Expressive\Container\ApplicationFactory::injectRoutesAndPipeline() must be of the type array, object given, called in /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-expressive/src/Container/ApplicationFactory.php on line 179 and defined in /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-expressive/src/Container/ApplicationFactory.php on line 191
```
Stack trace:
1. {main}() /home/travis/build/Codeception/Codeception/codecept:0
2. Codeception\Application->run() /home/travis/build/Codeception/Codeception/codecept:36
3. Symfony\Component\Console\Application->run() /home/travis/build/Codeception/Codeception/src/Codeception/Application.php:103
4. Symfony\Component\Console\Application->doRun() /home/travis/build/Codeception/Codeception/vendor/symfony/console/Application.php:122
5. Symfony\Component\Console\Application->doRunCommand() /home/travis/build/Codeception/Codeception/vendor/symfony/console/Application.php:191
6. Symfony\Component\Console\Command\Command->run() /home/travis/build/Codeception/Codeception/vendor/symfony/console/Application.php:846
7. Codeception\Command\Run->execute() /home/travis/build/Codeception/Codeception/vendor/symfony/console/Command/Command.php:264
8. Codeception\Command\Run->runSuites() /home/travis/build/Codeception/Codeception/src/Codeception/Command/Run.php:293
9. Codeception\Codecept->run() /home/travis/build/Codeception/Codeception/src/Codeception/Command/Run.php:366
10. Codeception\Codecept->runSuite() /home/travis/build/Codeception/Codeception/src/Codeception/Codecept.php:153
11. Codeception\SuiteManager->initialize() /home/travis/build/Codeception/Codeception/src/Codeception/Codecept.php:182
12. Codeception\Module\ZendExpressive->_initialize() /home/travis/build/Codeception/Codeception/src/Codeception/SuiteManager.php:80
13. Zend\ServiceManager\ServiceManager->get() /home/travis/build/Codeception/Codeception/src/Codeception/Module/ZendExpressive.php:59
14. Zend\ServiceManager\ServiceManager->create() /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:530
15. Zend\ServiceManager\ServiceManager->doCreate() /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:597
16. Zend\ServiceManager\ServiceManager->createFromFactory() /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:634
17. Zend\ServiceManager\ServiceManager->createServiceViaCallback() /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1094
18. call_user_func:{/home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:934}() /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:934
19. Zend\Expressive\Container\ApplicationFactory->__invoke() /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:934
20. Zend\Expressive\Container\ApplicationFactory->injectRoutesAndPipeline() /home/travis/build/Codeception/Codeception/frameworks-zend-expressive/vendor/zendframework/zend-expressive/src/Container/ApplicationFactory.php:179
```

The only interaction between ZendExpressive and my code at [this point](https://github.com/Codeception/Codeception/blob/2.2.9/src/Codeception/Module/ZendExpressive.php#L59) is
```php
$this->application = $this->container->get('Zend\Expressive\Application');
```
(It is line 12 in the stacktrace)

Is there something that I need to change in my application configuration to get it working?
The code is here: https://github.com/Naktibalda/codeception-zend-expressive-tests

**Comment by Rud5G** [comment]
I have this too. even when restarted with: 
`composer create-project zendframework/zend-expressive-skeleton <project-path>`

**Comment by Koopzington** [comment]
config.php returns an instance of ArrayObject
If you return $config directly, you should be good to go (provided you don't use Aura.Di judging by the comment)

**Comment by weierophinney** [comment]
Fixed with 1.1.1.
